charts/agent-operator/templates/operator-deployment.yaml

```diff
@@ -40,6 +40,10 @@ spec:
         resources:
         {{- toYaml . | nindent 10 }}
         {{- end }}
+        args:
+          {{- if .Values.kubeletService.enabled }}
+          - --kubelet-service={{ .Values.kubeletService.namespace }}/{{ template "ga-operator.fullname" . }}-kubelet
```

Comment:
Hmm, will `{{ template "ga-operator.fullname" . }}` produce an obvious value for users? Should we maybe just have the whole thing be configurable? Generally you'd probably want `kube-system/kubelet` here (or similar) just to make it easier to write ServiceMonitors that consume the service.

Reply:
(And maybe if the whole value is configurable, we can just check to see if it's non-empty instead of having an enabled field? WDYT?)

Reply:
> Hmm, will `{{ template "ga-operator.fullname" . }}` produce an obvious value for users? Should we maybe just have the whole thing be configurable? Generally you'd probably want `kube-system/kubelet` here (or similar) just to make it easier to write ServiceMonitors that consume the service.

I just followed the [Prom operator Helm chart](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml#L42) here. In [our own docs](https://grafana.com/docs/agent/latest/operator/getting-started/#install-agent-operator-on-kubernetes) we use `default/kubelet`

Reply:
I can make those changes tomorrow

Reply:
I think a loose "Helm convention" is to have things Helm manages also have the Helm release name in the K8s object name (in this case a Service). However I don't think it's that important, so I've gone ahead with the above changes.

Reply:
> I think a loose "Helm convention" is to have things Helm manages also have the Helm release name in the K8s object name (in this case a Service). However I don't think it's that important, so I've gone ahead with the above changes.

I think that's fair and completely agree FWIW, I'm just concerned about whether it'll be easy for users to find what the service name ends up being :)